index.js

```diff
@@ -477,18 +477,28 @@ module.exports = class Hyperdrive extends ReadyResource {
     await Promise.allSettled(proms)
   }
 
-  hasBlock (start, end) {
+  _hasBlock (start, end) {
     return this.blobs.core.has(start, end)
   }
 
-  async has (folder = '/') {
-    const entries = this.list(folder)
-    for await (const entry of entries) {
-      const b = entry.value.blob
-      const has = await this.hasBlock(b.blockOffset, b.blockOffset + b.blockLength)
-      if (!has) return false
+  async has (path = '/') {
+    if (!path) return false
+    await this.getBlobs()
+    const isFolder = path.endsWith('/')
```

Comment:
this is too brittle, just do an .entry lookup to see if it has an entry, if no entry -> assume its a folder